src/oic/oic/message.py

```diff
@@ -980,6 +1015,117 @@ class ResourceRequest(Message):
     "offline_access": [],
 }  # type: Dict[str, List[str]]
 
+# LOGOUT related messages
+
+SINGLE_OPTIONAL_JSON = ParamDefinition(dict, False, json_ser, json_deser, False)
+SINGLE_REQUIRED_JSON = ParamDefinition(dict, True, json_ser, json_deser, False)
+
+BACK_CHANNEL_LOGOUT_EVENT = "http://schemas.openid.net/event/backchannel-logout"
+
+
+class LogoutToken(Message):
+    """Defined in https://openid.net/specs/openid-connect-backchannel-1_0.html#LogoutToken ."""
+
+    c_param = {
+        "iss": SINGLE_REQUIRED_STRING,
+        "sub": SINGLE_OPTIONAL_STRING,
+        "aud": REQUIRED_LIST_OF_STRINGS,  # Array of strings or string
+        "iat": SINGLE_REQUIRED_INT,
+        "jti": SINGLE_REQUIRED_STRING,
+        "events": SINGLE_REQUIRED_JSON,
+        "sid": SINGLE_OPTIONAL_STRING,
+    }
+
+    def verify(self, **kwargs):
+        super().verify(**kwargs)
+
+        if "nonce" in self:
+            raise MessageException(
+                '"nonce" is prohibited from appearing in ' "a LogoutToken."
+            )
+
+        # Check the 'events' JSON
+        _keys = list(self["events"].keys())
+        if len(_keys) != 1:
+            raise ValueError('Must only be one member in "events"')
+        if _keys[0] != BACK_CHANNEL_LOGOUT_EVENT:
+            raise ValueError('Wrong member in "events"')
+        if self["events"][_keys[0]] != {}:
+            raise ValueError('Wrong member value in "events"')
+
+        # There must be either a 'sub' or a 'sid', and may contain both
+        if not ("sub" in self or "sid" in self):
+            raise ValueError('There MUST be either a "sub" or a "sid"')
+
+        try:
+            if kwargs["aud"] not in self["aud"]:
+                raise NotForMe("Not among intended audience")
+        except KeyError:
+            pass
+
+        try:
+            if kwargs["iss"] != self["iss"]:
+                raise NotForMe("Wrong issuer")
+        except KeyError:
+            pass
+
+        _now = utc_time_sans_frac()
+
+        _skew = kwargs.get("skew", 0)
+        _iat = self.get("iat", 0)
+
+        if _iat and _iat > (_now + _skew):
+            raise ValueError("Invalid issued_at time")
+
+        return True
+
+
+ID_TOKEN_VERIFY_ARGS = [
+    "keyjar",
+    "verify",
+    "encalg",
+    "encenc",
+    "sigalg",
+    "issuer",
+    "allow_missing_kid",
+    "no_kid_issuer",
+    "trusting",
+    "skew",
+    "nonce_storage_time",
+    "client_id",
+]
+
+
+class BackChannelLogoutRequest(Message):
+    """Defines the message used in https://openid.net/specs/openid-connect-backchannel-1_0.html ."""
+
+    c_param = {"logout_token": SINGLE_REQUIRED_STRING}
+
+    def verify(self, **kwargs):
+        super().verify(**kwargs)
+
+        args = {}
+        for arg in TOKEN_VERIFY_ARGS:
+            try:
```

Comment:
This can be rewritten using `get()` so no `try .. execept` is necessary.

Reply:
Well then you must add an `if .. then ..` so what's the benefit ?

Reply:
You could use a dict comprehension:

    args = {arg: kwargs[arg] for arg in TOKEN_VERIFY_ARG if arg in kwargs}

If you really want to shorten it, but i doubt it is more readable.

Reply:
You can update the args with `args.update(kwargs.get(arg, {})` which won't do anything in case of missing key.

Sorry, I sort of have a very strong aversion to `try .. except pass` statements and try to avoid them.

Reply:
`args.update(kwargs.get(arg, {}))` fails if kwargs[arg] actually exist.
@schlenk Yes, it's not more readable. But I like comprehensions. :-)

Reply:
Yeah, would be more complicated in this case, since you need a dict. Comprehension is fine :)